pkg/minikube/cluster/cluster.go

```diff
@@ -121,7 +121,7 @@ func StopHost(api libmachine.API) error {
 func DeleteHost(api libmachine.API) error {
 	host, err := api.Load(cfg.GetMachineName())
 	if err != nil {
-		return errors.Wrapf(err, "Error deleting host: %s", cfg.GetMachineName())
+		return errors.Wrapf(err, "Load: %s", cfg.GetMachineName())
```

Comment:
I'm wondering if we lose context here? I'd have to see the code to understand that this `Load: ...` error was at deletion time?

Reply:
In Go, in order to avoid duplication, the caller (one level up) adds the function name context. This effectively changes the error from:

"DeleteHost: Error deleting host: connection refused"

To:

"DeleteHost: Load: connection refused"

Note that there is actually more context in this case. The gist is that the function doesn't always know what name it was called with in the case of private or anonymous functions, but the caller always does. This was counter-intuitive to me at first, but it makes sense once you try to achieve human readable error message such as:

https://notes.shichao.io/gopl/ch5/#errors

Also:

https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully